packages/app/src/app/overmind/onInitialize.ts

```diff
@@ -108,25 +108,27 @@ export const onInitialize: OnInitialize = async (
   effects.preview.initialize(overmindInstance.reaction);
 
   // show terms message on first visit since new terms
-  if (!effects.browser.storage.get(seenTermsKey) && !state.isFirstVisit) {
```

Comment:
I wouldn't do this change in this function, since it's now very easy for people to add something unrelated after this body which will not get executed. This function itself is a function that initializes everything, adding an early return could cause confusion. It might be better to extract this logic into a separate function instead and add an early return there.

Reply:
I can create an `effect` instead if you want.
What namespace should I put it?

Or do you prefer just having the function inside the same file and passing `effects` & `state`?
```ts
const showPrivacyPolicyNotification = ({ effects, state }) => {
  if (effects.browser.storage.get(seenTermsKey) || state.isFirstVisit) {
    return;
  }
  
  // ...
}

// ...

showPrivacyPolicyNotification({ effects, state });
```

I prefer having an `effect`, since that will make typing a lot easier

Reply:
Yes, agree an `effect` or `action` would be best indeed! Maybe we can put it here? `packages/app/src/app/overmind/internalActions.ts`, then it's an internal action on the root.